Update SwanLake Beta 3 release notes with Ballerina OpenAPI Tool changes (#916)
ballerina-platform/ballerina-release: release-notes/swan-lake-beta3-release-note.md

#### New Features

##### Ballerina OpenAPI Tool

> - Introduced new command line option to generate all the record fields that are not specifically mentioned as 

Reviewer: ```suggestion
- Introduced a new command-line option to generate all the record fields that are not specifically mentioned as 
```

  `nullable:false` in openAPI schema property as nullable to reduce the type conversion errors in the OpenAPI to
  Ballerina command
  >`bal openapi -i <openapi-contract-file>  --nullable`
- Introduced new command line option for adding user required license or copyright headers for generated Ballerina

>   files in OpenAPI to Ballerina command

Reviewer: ```suggestion
  files via the OpenAPI to Ballerina command
```

  >`bal openapi -i <openapi-contract-file> --license <license-file> `
- Introduced a new command line option to generate the JSON file in the Ballerina to OpenAPI command
  >`bal openapi -i <service-file> --json`
- Added support to generate boilerplate of test functions for each remote function implemented within a
  client connector  
  
  

#### Improvements
##### Ballerina OpenAPI Tool
###### Ballerina OpenAPI client and schema generation improvements for the OpenAPI to Ballerina command
- Added support to generate suitable client connector authentication mechanisms by mapping with security schemes
  given in the OpenAPI specification (OAS)
- Added support to generate api documentation for client init method, remote functions and records

> - Added support to users to set common client configurations when initializing the connector 

Reviewer: ```suggestion
- Added support to facilitate users to set common client configurations when initializing the connector 
```

- Added support for generating records for nested referenced schemas in the OpenAPI specification 
- Added improvement to select `https` server URL when multiple URLs given in the OpenAPI specification
 
###### The Ballerina to OpenAPI command improvements

> - Added support for Language Server Extension 

Reviewer: ```suggestion
- Added support for the language server extension 
```

- Improved the response status code map to `202` when resource function hasn't `return` type in the resource function

> - Improved mapping different status code response in resource function

Reviewer: ```suggestion
- Improved mapping different status code responses in the resource function
```

- Enhanced generating openAPI schema with Ballerina typeInclusion scenarios

> - Added resource function api documentation mapping to OAS description and summary

Reviewer: ```suggestion
- Added resource function API documentation mapping to the OAS description and summary
```
